projects/plugins/jetpack/modules/masterbar/admin-menu/class-dashboard-switcher-tracking.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Quick switcher tracking file.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+use Automattic\Jetpack\Terms_Of_Service;
+use Automattic\Jetpack\Tracking;
+use Jetpack_Plan;
+
+/**
+ * Class Dashboard_Switcher_Tracking
+ */
+class Dashboard_Switcher_Tracking {
+
+	/**
+	 * Jetpack Tracking library will prefix the event name with "jetpack_*" automatically.
+	 */
+	const JETPACK_EVENT_NAME = 'dashboard_quick_switch_link_clicked';
+
+	const WPCOM_EVENT_NAME = 'wpcom_dashboard_quick_switch_link_clicked';
+	/**
+	 * Jetpack tracking object.
+	 *
+	 * @var Tracking
+	 */
+	private $tracking;
+
+	/**
+	 * Current site plan.
+	 *
+	 * @var string
+	 */
+	private $plan;
+
+	/**
+	 * The wpcom_tracks wrapper function.
+	 *
+	 * @var callable
+	 */
+	private $wpcom_tracking;
+
+	/**
+	 * Dashboard_Switcher_Tracking constructor.
+	 *
+	 * @param Tracking $tracking       Jetpack tracking object.
+	 * @param callable $wpcom_tracking A wrapper over wpcom event record.
+	 * @param string   $plan           The current site plan.
+	 */
+	public function __construct( Tracking $tracking, callable $wpcom_tracking, $plan ) {
+		$this->tracking       = $tracking;
+		$this->plan           = $plan;
+		$this->wpcom_tracking = $wpcom_tracking;
+	}
+
+	/**
+	 * Create an event for the Quick switcher when the user changes it's preferred view.
+	 *
+	 * @param string $screen The screen page.
+	 * @param string $view   The new preferred view.
+	 */
+	public function record_switch_event( $screen, $view ) {
+		$event_props = array(
+			'current_page' => $screen,
+			'destination'  => $view,
+			'plan'         => $this->plan,
+		);
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$event_props['blog_id'] = get_current_blog_id();
+			$wpcom_tracking         = $this->wpcom_tracking;
+			$wpcom_tracking( $event_props );
+		} else {
+			$this->record_jetpack_event( $event_props );
+		}
+	}
+
+	/**
+	 * Get the current site plan or 'N/A' when we cannot determine site's plan.
+	 *
+	 * @todo: This method can be reused as a wrapper over WPCOM and Atomic as way to get site's current plan (display name).
+	 *
+	 * @return string
+	 */
+	public static function get_plan() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			if ( class_exists( '\WPCOM_Store_API' ) ) {
+				// @todo: Maybe introduce a wrapper for this since we are duplicating it from WPCOM_Admin_Menu:253
+				$products = \WPCOM_Store_API::get_current_plan( \get_current_blog_id() );
+				if ( array_key_exists( 'product_name_short', $products ) ) {
+					return $products['product_name_short'];
+				}
+			}
+
+			return 'N/A'; // maybe we should return free or null? At the moment it's safe to return 'N/A' since we use it only for passing it to the event.
```

Comment:
I think `N/A` or `null` is fine here. Returning free will give us inaccurate data. It's better we are transparent if we don't know the plan.